File: socialByJava/src/main/java/com/example/socialbyjava/MVC/service/UserServiceImpl.java
package com.example.socialbyjava.MVC.service;


import com.example.socialbyjava.Entity.User;
import com.example.socialbyjava.MVC.repository.UserRepository;
import com.example.socialbyjava.etc.JwtUtil;
import com.example.socialbyjava.etc.response.UserResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Repository;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
@RequiredArgsConstructor
public class UserServiceImpl implements UserService {

    private final UserRepository userRepository;

    private final JwtUtil jwtUtil;
    @Override
    public UserResponse getUserInfo(String authorizationHeader) {
        UUID userId = UUID.fromString(jwtUtil.getUserIdFromToken(authorizationHeader));

        User user = userRepository.findByUserId(userId).get();

        UserResponse userResponse = UserResponse.builder()
                .name(user.getName())
                .provider(user.getProvider())
                .build();

        return userResponse;
    }
}

File: Spring/src/main/kotlin/com/example/spring/domain/user/service/UserServiceImpl.kt
package com.example.spring.domain.user.service


import com.example.spring.domain.user.dto.UserInfoDto
import com.example.spring.domain.user.entity.User
import com.example.spring.domain.user.repository.UserRepository
import com.example.spring.security.jwt.JwtUtil
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserServiceImpl(

    private val userRepository: UserRepository,
    private val jwtUtil: JwtUtil
) : UserService {


    override fun findUserInfo(authorizationHeader: String): UserInfoDto {
        var userId : String = jwtUtil.getUserIdFromToken(authorizationHeader)
        var user : User? = userRepository.findByUserId(UUID.fromString(userId))

        var userInfoDto : UserInfoDto = UserInfoDto(name = user!!.name, providerId = user.providerId, provider = user.provider,uuid = user.userId)

        return userInfoDto
    }
}